File: 프로그래머스/unrated/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;

class Solution {
    public int[] solution(int l, int r) {
        ArrayList<Integer> res = new ArrayList<>();
        for (int i = l; i <= r; i++) {
            boolean flag = true;
            int tmp = i;
            while (tmp > 0) {
                if (tmp % 10 != 0 && tmp % 10 != 5) {
                    flag = false;
                    break;
                }
                tmp /= 10;
            }
            if (flag) res.add(i);
        }

        int[] answer = new int[res.size()];
        for (int i = 0; i < res.size(); i++) {
            answer[i] = res.get(i);
        }
        
        if(answer.length == 0) {
            answer = new int[]{-1};
            return answer;
        } else {
            return answer;
        }
    }
}

File: 프로그래머스/unrated/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):
    answer = []
    for i in range(l, r+1):
        flag = True
        num = i
        while num:
            if num % 10 != 5 and num % 10 != 0:
                flag = False
                break
            num //= 10
            
        if flag: 
            answer.append(i)
        
    if len(answer): return answer
    else: return [-1]